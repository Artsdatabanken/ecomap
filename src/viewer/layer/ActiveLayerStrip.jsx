import React from 'react'
import { FlatButton, ListItem, Avatar } from 'material-ui'
import {hexToRgbaString} from '../../graphics/color/colorfunc'
import LayerPaintSettings from './settings/LayerPaintSettings'
import PaintSwatch from './PaintSwatch'

export default class ActiveLayerStrip extends React.Component {
  state = {expanded: false, showColorDialog: true}

  render () {
    const paint = this.props.paint
    return (
      <div>
        <ListItem
          disabled={false}
          primaryText={this.props.title}
          secondaryText={this.props.subTitle}
          leftAvatar={<Avatar src={this.props.imageUrl} />}
          rightIconButton={
            <PaintSwatch
              color={hexToRgbaString(paint.fillColor, paint.fillOpacity)}
              onClick={(e) => {
                e.stopPropagation()
                this.setState(prevState => ({showColorDialog: !prevState.showColorDialog}))
              }
              } />
      }
          onClick={() => this.setState(prevState => ({expanded: !prevState.expanded}))}
       />
        {this.state.expanded &&
          <div style={{ marginLeft: '24px', marginRight: '24px' }}>
            <div style={{position: 'relative', float: 'right'}}>
              <FlatButton label='Remove' onTouchTap={() => this.props.onDelete()} />
            </div>
            <LayerPaintSettings
              {...paint}
              onChange={this.props.onChange}
              onDelete={this.props.onDelete}
              showColorDialog={this.state.showColorDialog}
            />
            <div style={{ }}>
              <FlatButton label='Show correlation' onTouchTap={() => this.props.onAddLinkedLayer()} />
            </div>
          </div>
      }
      </div>
    )
  }
}
