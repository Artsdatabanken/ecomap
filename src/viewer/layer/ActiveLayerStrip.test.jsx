import React from 'react'
import renderer from 'react-test-renderer'
import muiTheme from '../../theme_jest'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

import ActiveLayerStrip from './ActiveLayerStrip'

it('renders ok', () => {
  const tree = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <ActiveLayerStrip paint={{fillColor: '#ff0044', fillOpacity: 0.1, radius: 0.2, coverage: 0.3}} />
    </MuiThemeProvider>
      )
  expect(tree.toJSON()).toMatchSnapshot()
})
