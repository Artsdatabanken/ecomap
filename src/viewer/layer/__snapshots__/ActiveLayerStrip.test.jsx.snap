// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ok 1`] = `
<div>
  <div>
    <span
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      open={null}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "backgroundColor": null,
          "border": 10,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "fontWeight": "inherit",
          "lineHeight": "16px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "zIndex": 1,
        }
      }
      tabIndex={0}
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          style={
            Object {
              "marginLeft": 0,
              "muiPrepared": true,
              "paddingBottom": 20,
              "paddingLeft": 72,
              "paddingRight": 56,
              "paddingTop": 20,
              "position": "relative",
            }
          }
        >
          <div
            className={undefined}
            size={40}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(188, 188, 188)",
                "borderRadius": "50%",
                "color": "#ffffff",
                "display": "inline-flex",
                "fontSize": 20,
                "height": 40,
                "justifyContent": "center",
                "left": 16,
                "muiPrepared": true,
                "position": "absolute",
                "top": 8,
                "userSelect": "none",
                "width": 40,
              }
            }
          />
          <div
            onClick={[Function]}
            style={
              Object {
                "backgroundImage": "linear-gradient(to bottom, rgba(255,0,68,0.1), rgba(255,0,68,0.1)), url(checkerboard.png)",
                "borderRadius": "50%",
                "height": "40px",
                "position": "absolute",
                "right": "14px",
                "top": "14px",
                "width": "40px",
              }
            }
          >
            Â 
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;
